### PR TITLE
Adding authentication

### PR DESCRIPTION
**Pull request by bruttozz**
This is under Shawn and Paul's guidance, Shawn said he thinks the method is correct but there still need both local and online tests, due to my issues when building Docker image, I'm not able to test it locally now and still waiting for Keval, need modifications after his work is done.
Thanks Shawn and Paul!

### CODE REVIEW

**Comment by SamiSousa** [comment]
It's failing on Travis due to the osb-broker-lib being out of date. We'll need to update some of our dependencies before we can merge this.

**Comment by SamiSousa** [comment]
I'll test this locally before we merge. Would be nice to have others test too.

**Comment by SamiSousa** [comment]
Encountering this issue:
```
root@sami-VirtualBox:/usr/local/go/src/github.com/dataverse-broker/dataverse-broker# make deploy-openshift
GOOS=linux GOARCH=amd64 CGO_ENABLED=0 \
go build -o dataverse-broker-linux --ldflags="-s" github.com/dataverse-broker/dataverse-broker/cmd/dataverse-broker
cp dataverse-broker-linux image/dataverse-broker
docker build image/ -t "quay.io/dataverse-broker/dataverse-broker:ddc6816"
Sending build context to Docker daemon  21.72MB
Step 1/4 : FROM fedora
 ---> 9110ae7f579f
Step 2/4 : COPY whitelist /opt/dataverse-broker/whitelist
 ---> Using cache
 ---> 9b60eb4e4454
Step 3/4 : ADD dataverse-broker /opt/dataverse-broker/dataverse-broker
 ---> Using cache
 ---> a0b59c576546
Step 4/4 : CMD /opt/dataverse-broker/dataverse-broker --port 8080 -v 4 --logtostderr
 ---> Using cache
 ---> 96248937b36c
Successfully built 96248937b36c
Successfully tagged quay.io/dataverse-broker/dataverse-broker:ddc6816
oc get project bdc || oc new-project bdc
NAME      DISPLAY NAME   STATUS
bdc                      Active
openshift/deploy.sh quay.io/dataverse-broker/dataverse-broker:ddc6816
make: execvp: openshift/deploy.sh: Permission denied
Makefile:38: recipe for target 'deploy-openshift' failed
make: *** [deploy-openshift] Error 127
```

Any idea how to fix this?

**Comment by SamiSousa** [comment]
Ran `chmod u+x openshift/deploy.sh` and `make deploy-openshift` runs error-free.

**Comment by SamiSousa** [comment]
Error with `make deploy-openshift` where `dataverse-broker-ssl` isn't created is fixed, but a new bug arises:
```
# oc logs -f deployments/dataverse-broker
I0424 01:16:39.167164       1 main.go:112] Starting broker!
I0424 01:16:39.173048       1 main.go:125] Starting secure broker with file based TLS cert and key
I0424 01:16:39.173057       1 server.go:110] Starting server on :8443
2018/04/24 01:16:52 http: TLS handshake error from 172.17.0.1:52994: EOF
...
```

Maybe this is due to the GetCatalog function still uses an http request?

**Comment by SamiSousa** [comment]
Also just noticed we added our own project to the vendors folder? That sounds like a recipe for disaster, so we should get rid of that.

File: Gopkg.lock [hunk 1]
```diff
@@ -7,6 +7,27 @@
   packages = ["quantile"]
   revision = "4c0e84591b9aa9e6dcfdf3e020114cd81f89d5f9"
 
+[[projects]]
+  branch = "master"
+  name = "github.com/dataverse-broker/dataverse-broker"
```
Comment (SamiSousa) [inline Gopkg.lock]: We should find a way to remove this as a dependency.

Comment (bruttozz) [reply Gopkg.lock]: Who added this in the Gopkg.lock?

Comment (SamiSousa) [reply Gopkg.lock]: This is autogenerated, so go must think we need this dependency.

Comment (kev5) [reply Gopkg.lock]: The only way is to probably to manually edit the Gopkg.toml file to remove the dependency. (https://github.com/golang/dep/issues/1470)

Comment (SamiSousa) [reply Gopkg.lock]: I checked the toml file, no mention of our repo.

File: Makefile [hunk 2]
```diff
@@ -35,8 +35,8 @@ deploy-helm: image ## Deploys image with helm
 	--set image="$(IMAGE):$(TAG)",imagePullPolicy="$(PULL)"
 
 deploy-openshift: image ## Deploys image to openshift
-	oc get project bdc || oc new-project bdc
-	oc process -f openshift/dataverse-broker.yaml -p IMAGE=$(IMAGE):$(TAG) | oc apply -f -
+	oc project bdc || oc new-project bdc
```
Comment (SamiSousa) [inline Makefile]: Here, the Makefile will switch you to the correct namespace on successful "oc project bdc"

File: openshift/dataverse-broker.yaml [hunk 3]
```diff
@@ -53,33 +96,62 @@ objects:
...
+- description: base64 encoded SSL CA Certificate
+  displayName: base64 encoded SSL CA Certificate
+  name: BROKER_CA_CERT
+  required: true
+  value:
+
+- description: Namespace that the starter pack is deployed to
+  displayName: Namespace
+  name: NAMESPACE
+  value: bdc
```
Comment (SamiSousa) [inline openshift/dataverse-broker.yaml]: This little guy is my contribution. We should also use this parameter anywhere else there's a mention of the namespace

File: openshift/dataverse-broker.yaml [hunk 4]
```diff
@@ -4,31 +4,73 @@ metadata:
...
+      kubernetes.io/service-account.name: dataverse-broker-client
+  type: kubernetes.io/service-account-token
+
 - apiVersion: servicecatalog.k8s.io/v1beta1
   kind: ClusterServiceBroker
   metadata:
     name: dataverse-broker
   spec:
-    url: http://dataverse-broker.bdc.svc.cluster.local
+    url: https://dataverse-broker.${NAMESPACE}.svc.cluster.local
```
Comment (SamiSousa) [inline openshift/dataverse-broker.yaml]: Using namespace parameter here makes it easier to modify the project name, by editing the value of the parameter.

**Comment by SamiSousa** [comment]
This is ready for another review @bruttozz

File: openshift/dataverse-broker.yaml (hunk 4 above)
Comment (bruttozz) [reply openshift/dataverse-broker.yaml]: Awesome! Creative! This fixes the issue?

Comment (SamiSousa) [reply openshift/dataverse-broker.yaml]: I set the value of the NAMESPACE parameter to bdc to match everything else.

**Pull request merged**